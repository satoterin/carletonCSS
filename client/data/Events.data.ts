import { Event } from '../types'

const EVENTS: Event[] = [
  {
    id: 1,
    title: 'Blockchain + Crypto Q&A Session',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 2,
    title: 'Become a Senior Engineer in 2 years',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 3,
    title: 'Blockchain and Cryptocurrency Landscape',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 4,
    title: 'A Chat With Ross Video',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 5,
    title: 'Vulnerabilities in Web Apps',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 6,
    title: 'Declassified Online School Survival Guide',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 7,
    title: 'Afternoon Among Us',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Social'
  },
  {
    id: 8,
    title: 'test',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 9,
    title: 'test',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 10,
    title: 'test',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 11,
    title: 'test',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  },
  {
    id: 12,
    title: 'test',
    body: 'test body',
    start: '2017-01-01T00:00:00.000Z',
    end: '2017-01-01T00:00:00.000Z',
    category: 'Workshop'
  }
]

export { EVENTS }
