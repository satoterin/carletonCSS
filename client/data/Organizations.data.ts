import { Organization } from '../types'

const ORGANIZATIONS: Organization[] = [
  {
    id: 1,
    title: 'Carleton Computer Science Society',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/images/ccss-logo.png'
  },
  {
    id: 2,
    title: 'CuHacking',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/goodcuHacking.png'
  },
  {
    id: 3,
    title: 'SCESoc',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/scesoc.png'
  },
  {
    id: 4,
    title: 'Women in Computer Science',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/wics.png'
  },
  {
    id: 5,
    title: 'Carleton A.I. Society',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/carletonai.png'
  },
  {
    id: 6,
    title: 'CU Blueprint',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/blueprint.jpg'
  },
  {
    id: 7,
    title: 'Technolgap',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/technolgap.png'
  },
  {
    id: 8,
    title: 'Carleton Web Development Club',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/webdev.png'
  },
  {
    id: 9,
    title: 'Carleton eggX',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/eggx-long.png'
  },
  {
    id: 10,
    title: 'BITsoc',
    description: 'test',
    imgUrl: 'https://bitsoc.ca/assets/Logo.png'
  },
  {
    id: 11,
    title: 'Carleton Cyber Security Club',
    description: 'test',
    imgUrl: 'https://carletoncybersecurityclub.github.io/assets/images/1.png'
  },
  {
    id: 12,
    title: 'Game Design Carleton',
    description: 'test',
    imgUrl: 'https://ccss.carleton.ca/ccss-website/images/blueprint.jpg'
  }
]

export { ORGANIZATIONS }
