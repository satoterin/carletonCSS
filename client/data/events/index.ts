export { exampleEvents } from './Events.example'
export { getEvents } from './Events.request'
