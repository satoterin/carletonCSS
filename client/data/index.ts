export * from './Events.data'
export * from './Organizations.data'
