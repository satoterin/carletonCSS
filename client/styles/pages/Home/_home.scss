@import './EventFilter/event-filter';
@import './HomeHeader/home-header';

@media only screen and (max-width: 600px) {
  .home-h1 {
    display: none;
  }
}
