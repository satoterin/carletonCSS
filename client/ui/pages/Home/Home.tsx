import React from 'react'
import { Event } from '../../../types'
import { EventCard } from '../../shared'
import { HomeHeader } from './HomeHeader'

interface Props {
  data: Event[];
}

const Home = ({ data }: Props) => {
  return (
    <div className="home-page">
      <HomeHeader/>
      <div className="page-wrap">
        <div className="card-container">
          {
            data.map((event, index) => (
              <EventCard key={index} title={event.title} date="27 May 2021" body={event.body}/>
            ))
          }
        </div>
      </div>
    </div>
  )
}

export { Home }
