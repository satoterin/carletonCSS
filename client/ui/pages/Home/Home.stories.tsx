import { storiesOf } from '@storybook/react'

import { Home } from './Home'
import { exampleEvents } from '../../../data'

storiesOf('Home', module).add('Page', () => {
  return <Home data={exampleEvents}/>
})
