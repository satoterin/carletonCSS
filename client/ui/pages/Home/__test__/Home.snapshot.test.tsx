import React from 'react'
import renderer from 'react-test-renderer'

import { Home } from '../Home'
import { exampleEvents } from '../../../../data'

it('renders correctly', () => {
  const tree = renderer.create(<Home data={exampleEvents}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
