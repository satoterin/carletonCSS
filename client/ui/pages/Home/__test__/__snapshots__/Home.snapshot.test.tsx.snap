// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h1>
  Welcome to My Next App!
</h1>
`;
